Examples/ObjectiveC/UserTrackingModesExample.m

```diff
@@ -3,23 +3,182 @@
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, max * 0.65)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.9, max)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, 0)];
+    [bezierPath closePath];
+    
+    return bezierPath.CGPath;
+}
+
+-(void)updateArrow:(MGLUserTrackingMode)mode {
+    CGColorRef arrowStrokeColor;
+    CGPoint arrowPosition;
+    CGColorRef arrowFillColor;
+    CGFloat arrowRotation;
+    
+    switch (mode) {
+        case MGLUserTrackingModeNone:
+            arrowStrokeColor = [[UIColor whiteColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+            arrowFillColor = [[UIColor clearColor] CGColor];
+            arrowRotation = 0;
+            break;
+        case MGLUserTrackingModeFollow:
+            arrowStrokeColor = self.tintColor.CGColor;
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2 + 2, UserLocationButtonSize / 2 - 2);
+            arrowFillColor = [[UIColor clearColor] CGColor];
+            arrowRotation = 0.66;
+            break;
+        case MGLUserTrackingModeFollowWithHeading:
+            arrowStrokeColor = [[UIColor clearColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2 + 2, UserLocationButtonSize / 2 - 2);
+            arrowFillColor = self.tintColor.CGColor;
+            arrowRotation = 0.66;
+            break;
+        case MGLUserTrackingModeFollowWithCourse:
+            arrowStrokeColor = [[UIColor clearColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+            arrowFillColor = self.tintColor.CGColor;
+            arrowRotation = 0;
+            break;
+    }
+    
+    [_arrow setFillColor:arrowFillColor];
+    [_arrow setStrokeColor:arrowStrokeColor];
+    [_arrow setAffineTransform:CGAffineTransformMakeRotation(arrowRotation)];
+    
+    [self layoutIfNeeded];
+}
+
+@end
+
+@interface UserTrackingModesExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) UserLocationButton IBOutlet *button;
+
+@end
+
 @implementation UserTrackingModesExample
 
 - (void)viewDidLoad {
     [super viewDidLoad];
 
-    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle darkStyleURL]];
-    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle darkStyleURL]];
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView.delegate = self;
 
-    mapView.userTrackingMode = MGLUserTrackingModeFollowWithHeading;
+    self.mapView.tintColor = [UIColor redColor];
+    self.mapView.attributionButton.tintColor = [UIColor lightGrayColor];
 
-    // The user location annotation takes its color from the map view tint.
```

Comment:
I think these comments about user location and attribution button tint color clarify a behavior that’s not immediately obvious — can we add them back?